src/System.IO.Ports/src/System/IO/Ports/SerialStream.Unix.cs

```diff
@@ -441,55 +522,49 @@ internal unsafe void Write(byte[] array, int offset, int count, int timeout)
                 throw new ArgumentOutOfRangeException(nameof(offset), SR.ArgumentOutOfRange_NeedPosNum);
             if (count < 0)
                 throw new ArgumentOutOfRangeException(nameof(count), SR.ArgumentOutOfRange_NeedPosNum);
-            if (count == 0) return; // no need to expend overhead in creating asyncResult, etc.
             if (array.Length - offset < count)
                 throw new ArgumentException(SR.Argument_InvalidOffLen);
```

Comment:
Can all of this argument validation (here and elsewhere) be shared between the Windows and Unix implementations?